memory/combat_manager.py

```diff
@@ -0,0 +1,150 @@
+from enum import Enum
+
+from memory.core import mem_handle
+
+
+class CombatDamageType(Enum):
+    NONE = 0
+    _unknown = 1
+    Sword = 2
+    _unknown2 = 3
+    _unknown3 = 4
+    _unknown4 = 5
+    _unknown5 = 6
+    _unknown6 = 7
+    Moon = 8
+
+
+class CombatSpellLock:
+    def __init__(self):
+        self.name_loc_id = None
+        self.damage_type = CombatDamageType.NONE
+
+
+class CombatCastingData:
+    def __init__(self):
+        self.spell_locks = []
+
+
+class CombatEnemy:
+    def __init__(self, params=dict):
+        self.max_hp = None
+        self.dead = params["dead"]
+        self.current_hp = params["current_hp"]
+        self.speed = None
+        self.base_physical_defense = None
+        self.base_physical_attack = None
+        self.base_magic_attack = None
+        self.casting_data = CombatCastingData()
+
+
+class CombatPlayer:
+    def __init__(self, params=dict):
```

Comment:
any particular reason for the dict approach as opposed to named parameters?

Reply:
mostly was just convenient for now, will move to named params